### PR TITLE
TSC project roles documentation

### PR DESCRIPTION
**Pull request by ioannisg**
Uploading the Zephyr Project Roles Document that has been voted at TSC on Feb 5th, 2020.
The uploaded content is **identical to what was voted,** except for
- the Charter Changes (that do not need to be checked-in the Zephyr documentation).
- the opens and plans for the future (not applicable to be documented)
- the current Charter (not applicable for Zephyr documentation)

Link to the voted Project Roles proposal: https://docs.google.com/presentation/d/15k2yLWL6KeisNfhso4-tMIY9MT1jjJGUpAHRGl83w6g/edit#slide=id.p

### CODE REVIEW

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: why orphan? I would make sure this is linked in the table of contents somewhere instead of just via a reference.

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: no need for this top line

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: no need for top line

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: this one still have committer terminology in it.

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: need to convert this into a RST table so we can update, picture wont do.

File: doc/contribute/project_roles.rst
Comment (nashif) [inline doc/contribute/project_roles.rst]: I would leave Opens and Plan out of this.

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: It is linked in the index.rst of contribute/ section

**Review by ioannisg — COMMENTED** [review]
@nashif I will go through the changes you requested; for now, I want you to confirm you are OK with where in the documentation I placed the proposal.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Yes, could you fix this one @nashif ? I don't have the source of this figure.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Fixed

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Fixed

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Yes, this is fine with me too. Since TSC voted for this, I wanted to include this in, but agree it does not make much sense to document the open questions and our future plans.

Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Removed.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Yes, agreed, done.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: now present also in  the table of contents

File: doc/contribute/project_roles.rst
Comment (aescolar) [inline doc/contribute/project_roles.rst]: This was discussed in the TSC as problematic for areas with too few involved parties. If we write it down here, some will forget about the context and try to follow it to the letter of the law blocking progress in those areas.

File: doc/contribute/project_roles.rst
Comment (aescolar) [inline doc/contribute/project_roles.rst]: Would you consider linking to https://docs.zephyrproject.org/latest/development_process/dev_env_and_tools.html#give-reviewers-time-to-review-before-code-merge ?

File: doc/contribute/project_roles.rst
Comment (aescolar) [inline doc/contribute/project_roles.rst]: To be clearer, Should this say "The involved parties and the assignee"?

File: doc/contribute/project_roles.rst
Comment (aescolar) [inline doc/contribute/project_roles.rst]: Would it make sense linking to https://github.com/zephyrproject-rtos/zephyr/wiki/Program-Management ?

File: doc/contribute/project_roles.rst
Comment (aescolar) [inline doc/contribute/project_roles.rst]: I did not understand this was voted on in the TSC..

**Review by ioannisg — COMMENTED** [review]
@aescolar thanks for looking at the PR, much appreciated.
For now, I think we should try to merge this PR as is, since this is *exactly* what the TSC has voted for. 
We are currently discussing to amend the proposal, in the Process Forum meeting, so I hope to be able to include your feedback in the next round.

For now, I would ask you and  @nashif  to double check that the content in this PR matches that of the voted document :)

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: https://docs.google.com/presentation/d/15k2yLWL6KeisNfhso4-tMIY9MT1jjJGUpAHRGl83w6g/edit#slide=id.p

Is this illustrated on Slide 23?

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: I think this can be appended, yes.
I'll add this on the amended proposal.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Agreed. Moving this to the amendment.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: i think the alternative option of any other maintainer should address the vast majority of these cases.
Anyways, I will bring this up while discussing the updated proposal.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: I think yes, this is not a problem. Does not modify the content. Will do.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Yes, that is fine to add. Will do.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: this is fixed now

**Comment by ioannisg** [comment]
@galak @carlescufi @MaureenHelm could you also take a look here?

**Review by nashif — APPROVED** [review]

File: doc/contribute/project_roles.rst
Comment (ioannisg) [inline doc/contribute/project_roles.rst]: will fix this.

File: doc/contribute/project_roles.rst
Comment (nashif) [reply doc/contribute/project_roles.rst]: this was discussed in the TSC and it should indeed be subsystem/area hierarchy.

Comment (ioannisg) [reply doc/contribute/project_roles.rst]: done now

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Fixed now

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Content looks good. Sorry to nitpick, but could you please update the commit message to say this content was approved by the TSC **on Feb 5, 2020**

**Comment by ioannisg** [comment]
> ould you please update the

Of course, no problem

**Review by MaureenHelm — APPROVED** [review]

File: doc/contribute/project_roles.rst
Comment (carlescufi) [inline doc/contribute/project_roles.rst]: I strongly recommend switching to the other table format:

```
.. list-table::
   :header-rows: 1

   * - ..
     - ..
     - **admin**

   * - Main Roles
     - Contributor
     - 

```

```

**Review by carlescufi — APPROVED** [review]
LGTM, thank you

File: doc/contribute/project_roles.rst
Comment (jhedberg) [inline doc/contribute/project_roles.rst]: Should we use here the precise names of the files like you're doing in the heading of this section, i.e. `MAINTAINERS` and `CODEOWNERS`?

File: doc/contribute/project_roles.rst
Comment (jhedberg) [inline doc/contribute/project_roles.rst]: This is really minor, but could you try to be consistent with the line lengths. Some lines you're splitting very short and then some (like these) letting be long. This should be simple to automate with most editors.

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: @carlescufi Zephyr Documentation accepts this table format, check here:

https://docs.zephyrproject.org/latest/guides/documentation/index.html

and the given link:
https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#grid-tables

Actually, the current table format is the best option to get all this complex merged rows and columns

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: I am going to try. I think I had seen some issues with sphinx not understanding the line breaks is two-depth indentation bullet-point listing

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Yes, that's a good point, actually, and does not change the content. I will apply the change.

Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Fixed now

File: doc/contribute/project_roles.rst
Comment (ioannisg) [reply doc/contribute/project_roles.rst]: Fixed, that too - let's wait to see if there are any doc build-issues

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
